Command.run: remove comment about API

emillon opened the pull request:
After trying to change the type, it turns out that adding `S` everywhere is not going in the right direction.

emillon commented:
I wanted to address the TODO item as a PR. Personally I think it's good to use `Args.t` instead of `Args.t list` and list manipulation, but if we don't think it's the right solution I'm fine with just removing the TODO item instead.

Alizter commented on src/dune_rules/command.mli:
```diff
@@ -72,21 +72,19 @@ module Args : sig
   -> ?stdout_to:Path.Build.t
   -> Action.Prog.t
-  -> Args.any Args.t list
+  -> Args.any Args.t
```
I definitely prefer this as it now encourages the right way to think about arguments. It's pretty evident that everybody was confused whether to use `Args.t list` or `Args.t` before.

Alizter replied:
But maybe I spoke to soon since it makes calling `run` almost always more awkward.

rgrinberg commented:
I don't think this is a good change tbh. The point of taking the list was to be symmetric with `Process.run`.

We can see this by how much noise this PR adds.

rgrinberg commented:
Yes, that sounds good.

emillon committed:
Command.run: remove comment about API

After trying to change the type, it turns out that adding `S` everywhere
is not going in the right direction.

Signed-off-by: Etienne Millon <me@emillon.org>
